### PR TITLE
Resource-related classes refactored

### PR DESCRIPTION
**Pull request by mfranciszkiewicz**
- `AbstractResourceManager` logic (partially) cleaned up and moved to `HyperdriveResourceManager`
- Classes accompanying `AbstractResourceManager` were moved to separate source files
- `ClientHandler` logic greatly simplified
- `ClientCommands` removed
- `FileResource`, `ResourceBundle`, etc. are now a single `Resource` class
- Removed extra parameters in `HyperdriveResourceManager.[add_task|add_file|add_files]` methods

### CODE REVIEW

**Comment by etam** [comment]
This is so big I don't even know where to start reading.

**Comment by mfranciszkiewicz** [comment]
@etam “Needs review” was removed

File: golem/resource/hyperdrive/resourcesmanager.py [hunk 1]
```diff
@@ -1,175 +1,240 @@
...
 
-        for entry in iterator:
-            files = [os.path.join(*split) for split in entry[1] if split]
-            if not files:
-                logger.debug("Received an empty file list for hash %r",
-                             entry[0])
-                continue
-            results.append([entry[0], files])
+        if len(results) != len(serialized):
+            logger.debug("Errors occurred while deserializing %r", serialized)
```
Comment (Elfoniok) [inline golem/resource/hyperdrive/resourcesmanager.py]: Is it so common? Maybe a warning here?

File: golem/resource/hyperdrive/resourcesmanager.py [hunk 2]
```diff
@@ -1,175 +1,240 @@
...
+        return self._add_task(*args)
+
+    def remove_task(self, task_id):
+        resources = self.storage.cache.remove(task_id)
+        if not resources:
             return
 
-        files = {path: self.storage.relative_path(path, task_id)
-                 for path in files}
+        client = self.new_client()
```
Comment (Elfoniok) [inline golem/resource/hyperdrive/resourcesmanager.py]: Quite strange, wouldn't it be better to simply hold reference to HyperdriveClient?

File: golem/resource/hyperdrive/resourcesmanager.py [hunk 3]
```diff
@@ -1,175 +1,240 @@
...
 
         if not all(checked.values()):
             missing = [f for f, exists in files.items() if not exists]
             logger.error("Resource manager: missing files (task: %r):\n%s",
                          task_id, missing)
-            return
+            return None, None
 
-        client = client or self.new_client()
+        client = self.new_client()
```
Comment (Elfoniok) [inline golem/resource/hyperdrive/resourcesmanager.py]: again why create it over and over?

**Comment by Elfoniok** [comment]
Very good change. It greatly simplifies resources code and its working properly.

File: golem/resource/hyperdrive/resourcesmanager.py (hunk 1 above)
Comment (mfranciszkiewicz) [reply golem/resource/hyperdrive/resourcesmanager.py]: Good point

File: golem/resource/hyperdrive/resourcesmanager.py (hunk 2 above)
Comment (mfranciszkiewicz) [reply golem/resource/hyperdrive/resourcesmanager.py]: `client` was created every time to support client configuration change on the fly. That is not done in Golem (now) and probably won't be.

Comment (mfranciszkiewicz) [reply golem/resource/hyperdrive/resourcesmanager.py]: I'll change the code to use a single `Client` instance.

**Review by Elfoniok — APPROVED** [review]

**Pull request merged**